### PR TITLE
Move `Deliver` OTEL activity to consumer dispatchers

### LINKED ISSUE #1621: RabbitMQActivitySource.Deliver cannot be used by types that implement Consumer
### Describe the bug

Currently ``RabbitMQActivitySource.Deliver`` is only called in the derived types ``AsyncEventingBasicConsumer`` and ``EventingBasicConsumer`` and  the method itself is marked as internal.

https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/89d472d8995b815ae5a05b9d998853945c428b99/projects/RabbitMQ.Client/client/events/AsyncEventingBasicConsumer.cs#L99-L105

This means, if a dev is deriving from the base types ``DefaultBasicConsumer``  or ``AsyncDefaultBasicConsumer`` or implements the appropriate interfaces, the Deliver method can't be invoked.

All the other relevant ActivitySource methods are called from within ChannelBase, so these should be fine.

I think the easiest way forward is to make a public RabbitMQActivitySource.Deliver method with the args from the eventArgs:

Current:
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/89d472d8995b815ae5a05b9d998853945c428b99/projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs#L122-L142

Proposed change:

```csharp
        internal static Activity Deliver(BasicDeliverEventArgs deliverEventArgs)
        {
            return Deliver(deliverEventArgs.RoutingKey, deliverEventArgs.Exchange, deliverEventArgs.DeliveryTag,
                deliverEventArgs.BasicProperties, deliverEventArgs.Body.Length);
        }

        public static Activity Deliver(string routingKey, string exchange, ulong deliveryTag,
            in ReadOnlyBasicProperties readOnlyBasicProperties, int bodySize)
        {
            if (!s_subscriberSource.HasListeners())
            {
                return null;
            }

            // Extract the PropagationContext of the upstream parent from the message headers.
            Activity activity = s_subscriberSource.StartLinkedRabbitMQActivity(
                UseRoutingKeyAsOperationName ? $"{routingKey} deliver" : "deliver",
                ActivityKind.Consumer, ContextExtractor(readOnlyBasicProperties));
            if (activity != null && activity.IsAllDataRequested)
            {
                PopulateMessagingTags("deliver", routingKey, exchange,
                    deliveryTag, readOnlyBasicProperties, bodySize,
                    activity);
            }

            return activity;
        }
```

Then the dev can call it in their own implementation of ``HandleBasicDeliver``. I don't see any easy way to do it automatically, like for the other methods, that would be preferable.

### Reproduction steps

-

### Expected behavior

-

### Additional context

Maybe @stebet  has a better idea, he wrote the original PR.

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #1621

### CODE REVIEW

**Comment by lukebakken** [comment]
cc @bollhals and @Tornhoof!!! 😸

**Comment by lukebakken** [comment]
@stebet - when you have a chance, take a look. Thanks!

File: projects/Test/OAuth2/OAuth2.csproj [hunk 1]
```diff
@@ -29,7 +29,7 @@
     <PackageReference Include="xunit" />
     <PackageReference Include="xunit.runner.visualstudio" PrivateAssets="all" />
     <PackageReference Include="System.Net.Http" />
-    <PackageReference Condition="'$(TargetFramework)'=='net472'" Include="System.Text.Json" />
+    <PackageReference Condition="'$(TargetFramework)'=='net472'" Include="System.Text.Json" NoWarn="NU1903" />
```
Comment (Kielek) [inline projects/Test/OAuth2/OAuth2.csproj:32]: Same here.

Comment (lukebakken) [reply projects/Test/OAuth2/OAuth2.csproj]: Thanks for noticing

**Comment by lukebakken** [comment]
Thanks everyone.

**Pull request merged**